pkg/vgmanager/wipe_devices.go

```diff
@@ -0,0 +1,85 @@
+package vgmanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	lvmv1alpha1 "github.com/openshift/lvm-operator/api/v1alpha1"
+	"github.com/openshift/lvm-operator/pkg/lsblk"
+	"github.com/openshift/lvm-operator/pkg/lvm"
+	"github.com/openshift/lvm-operator/pkg/wipefs"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func (r *VGReconciler) wipeDevicesIfNecessary(ctx context.Context, volumeGroup *lvmv1alpha1.LVMVolumeGroup, vgs []lvm.VolumeGroup, blockDevices []lsblk.BlockDevice) (bool, error) {
+	if volumeGroup.Spec.DeviceSelector == nil || !volumeGroup.Spec.DeviceSelector.ForceWipeDevicesAndDestroyAllData {
+		return false, nil
+	}
+
+	allPaths := append(volumeGroup.Spec.DeviceSelector.Paths, volumeGroup.Spec.DeviceSelector.OptionalPaths...)
+	wiped := false
+	for _, path := range allPaths {
+		if isDeviceAlreadyPartOfVG(vgs, path, volumeGroup) {
```

Comment:
I believe if the device is part of the vg and the vg was already creatd once and is discoverable in the node status, we can just skip this path. However, if the device is part of a vg that is not in the node status, we have to decide if we should risk killing the existing vg that is not managed by us by wiping the disk, or even erroring. WDYT?

Reply:
I've changed the implementation to check `isDeviceAlreadyPartOfVG` by verifying the existence from the `LVMVolumeGroupNodeStatus` instead of from `vgs` call. This will prevent a case where the name of a user-created volume group is the same as the one created in LVMS, and we assume that it is created by LVMS, and skip these devices.

Reply:
If we force wipe devices, shouldn't we also force wipe the devices that were created by LVMS and leftover?